tests/coalaHelperTest.py

```diff
@@ -0,0 +1,58 @@
+import json
+import os
+import shutil
+import unittest
+
+from coalahtml.helper import (get_file, parse_file_dict, build_file_graph,
+                              create_dir, copy_files, get_args)
+
+
+class coalaHelperTest(unittest.TestCase):
+
+    def setUp(self):
+        self.dir_name = "someRandomName"
+        self.test_dir_path = os.path.abspath(self.dir_name)
+        self.file_name = self.dir_name[::2]
+        test_section = 'default'
+        test_file_name = 'tempTest.py'
+        test_file_content = 'import this\n'
+        self.test_file_dict = {test_section: {
+            test_file_name: (test_file_content)}}
+        self.test_file_data = {test_file_name: test_file_content}
+
+    def tearDown(self):
+        pass
+
+    def test_get_file(self):
+        test_file_path = os.path.abspath(os.path.join(self.test_dir_path,
+                                                      self.file_name))
+        return_file_path = get_file(self.file_name, self.test_dir_path)
+        self.assertEqual(test_file_path, return_file_path)
+
+    def test_parse_file_dict(self):
+
+        result_file_data = parse_file_dict(self.test_file_dict)
+        self.assertEqual(result_file_data, self.test_file_data)
+
+    def test_create_dir(self):
+        self.assertEqual(create_dir(self.test_dir_path),
+                         self.test_dir_path)
+        os.rmdir(self.test_dir_path)
+
+    def test_copy_files(self):
+        s_dir = create_dir(os.path.abspath('S99'))
+        d_dir = create_dir(os.path.abspath('D99'))
```

Comment:
I had same thought at first, but I already have a method that does the same
thing. What's wrong?
ᐧ

On Fri, May 27, 2016 at 2:36 PM, Attila <notifications@github.com> wrote:

> In tests/coalaHelperTest.py
> <https://github.com/coala-analyzer/coala-html/pull/46#discussion_r64877363>
> :
>
> > +        return_file_path = get_file(self.file_name, self.test_dir_path)
> > +        self.assertEqual(test_file_path, return_file_path)
> > +
> > +    def test_parse_file_dict(self):
> > +
> > +        result_file_data = parse_file_dict(self.test_file_dict)
> > +        self.assertEqual(result_file_data, self.test_file_data)
> > +
> > +    def test_create_dir(self):
> > +        self.assertEqual(create_dir(self.test_dir_path),
> > +                         self.test_dir_path)
> > +        os.rmdir(self.test_dir_path)
> > +
> > +    def test_copy_files(self):
> > +        s_dir = create_dir(os.path.abspath('S99'))
> > +        d_dir = create_dir(os.path.abspath('D99'))
>
> Can you use tempfile for this?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-html/pull/46/files/45b9d8e02c8e3c164f75093a3f89146b39bb2e61..f1d171ab1b617f884f1e0d2e8afaebd9aaca398a#r64877363>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHDgOTCvoiUJG77CXSp-b2-8lDAqYghmks5qFrQzgaJpZM4Ie05w>
> .
>